components/stock-card/index.js

```diff
@@ -1,25 +1,51 @@
 import path from 'path';
 import Image from 'next/image';
 import StockOperation from '@components/stock-operation';
+import { useRouter } from 'next/router';
 
 export const Card = ({ stock }) => {
+  const router = useRouter();
+
+  const getStockOperation = () => {
+    return router.pathname == '/trading' ? (
+      <StockOperation
+        stockId={stock.stockId}
+        name={stock.name}
+        price={stock.price}
+        availableQty={stock.quantity}
+      />
+    ) : (
+      <StockOperation
+        stockId={stock.stockId}
+        name={stock.stockName}
+        price={stock.initialStockValue}
+        availableQty={stock.quantity}
+      />
```

Comment:
We should be able to reduce the above code to something simpler:

```js
const { stockId, stockName, initialStockValue, price, quantity } = stock;
const stockOperationPrice =  ...; // price or initialStockValue 
```

But, before that, can we connect to discuss how the sell / buy page works. I am not sure if we should add any logic that's dependent on the state of the route. This keeps us from creating a better pure component.

Reply:
Sure! Thanks for the review. 😄 🎉